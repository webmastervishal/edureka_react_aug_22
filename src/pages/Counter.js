import React, { useState, useEffect, useContext, useRef, useMemo } from "react";
import Layout from "./../layout";
import Cookies from "js-cookie";
import UserContext from "../context/UserContext";

function Counter() {
  const [count, setCount] = useState(0);
  const [numbers, setNumbers] = useState([1, 2, 3, 4, 4]);
  const [show, setShow] = useState(true);
  const [currencies, setCurrencies] = useState([]);
  const data = useContext(UserContext);
  const inputRef = useRef();

  async function fetchAPI(path) {
    const res = await fetch(
      `https://webmaster-fake-api.herokuapp.com/currencies`,
      {
        method: "GET",
        headers: {
          Authorization: "Bearer " + Cookies.get("token"),
        },
      }
    );
    const result = await res.json();
    setCurrencies(result);
  }

  useEffect(() => {
    // console.log("Mount/Update");
    fetchAPI();

    inputRef.current.style.height = "25px";
    inputRef.current.focus();

    //open ws connection to server
    // let time = 1;

    // const timer = setInterval(() => {
    //   time = time + 1;
    //   console.log("timer...", time);
    // }, 1000);

    // return () => {
    //   //close ws connection
    //   clearInterval(timer);
    //   console.log("Unmounting");
    // };
  }, []);

  useEffect(() => {
    //hit the api to fetch the user data : 1
    // console.log("calling second useeffect on count change");
  }, [count]);

  function displayMessage() {
    console.log("calling display message....");
    return <h1>Displaying message....</h1>;
  }
  return (
    <Layout>
      {show && (
        <>
          <h1 id="counter">Count: {count}</h1>
          <button id="increment" onClick={() => setCount(count + 1)}>
            +
          </button>
          <button id="decrement" onClick={() => setCount(count - 1)}>
            -
          </button>
        </>
      )}
      <button onClick={() => setShow(!show)}>{show ? "Hide" : "Show"}</button>
      {/* <h1>Numbers</h1>
      {numbers.map((num) => (
        <p>{num}</p>
      ))} */}
      {currencies.map((currency, index) => (
        <p key={index}>{currency.name}</p>
      ))}
      <input ref={inputRef} type="text" name="test" />

      {useMemo(() => displayMessage(), [])}
    </Layout>
  );
}

// class Counter extends React.Component {
//   constructor() {
//     super();

//     this.state = { count: 0 };
//     this.handleIncrement = this.handleIncrement.bind(this);
//     this.handleDecrement = this.handleDecrement.bind(this);
//   }

//   static getDerivedStateFromProps() {
//     //lifecycle method
//     //make api call
//   }

//   handleIncrement() {
//     this.setState({
//       count: this.state.count + 1,
//     });
//     // console.log("this", this);
//   }

//   handleDecrement() {
//     this.setState({
//       count: this.state.count - 1,
//     });
//   }

//   render() {
//     return (
//       <Layout>
//         <h1>Count: {this.state.count}</h1>
//         <button onClick={this.handleIncrement}>+</button>
//         <button onClick={this.handleDecrement}>-</button>
//       </Layout>
//     );
//   }

//   componentDidMount() {
//     //make api calls
//     //hitting api to fetch user id : 1
//     //open ws connection
//   }

//   componentDidUpdate() {
//     //hitting api to fetch user id: 2
//   }

//   componentWillUnmount() {
//     //close ws connection
//   }
// }

export default Counter;
