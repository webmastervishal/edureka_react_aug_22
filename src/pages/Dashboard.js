import React from "react";
import Layout from "./../layout";

class Dashboard extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Layout>
        <h1>Dashboard page</h1>
      </Layout>
    );
  }
}

export default Dashboard;
