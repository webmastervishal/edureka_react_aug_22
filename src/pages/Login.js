import React from "react";
import "./login.css";
import Cookies from "js-cookie";

class Login extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      password: "",
      error: "",
    };

    const isAuthenticated = !!Cookies.get("token");
    if (isAuthenticated) {
      props.history.push("/dashboard");
    }
  }

  handleEmail = (e) => {
    this.setState({ ...this.state, email: e.target.value });
  };

  handlePassword = (e) => {
    this.setState({ ...this.state, password: e.target.value });
  };

  handleLogin = async (e) => {
    e.preventDefault();
    console.log("state", this.state);

    //hit the login api

    const res = await fetch(
      "https://webmaster-fake-api.herokuapp.com/auth/login",
      {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          email: this.state.email,
          password: this.state.password,
        }),
      }
    );

    const result = await res.json();
    console.log("result", result);

    if (result.status != 401) {
      //if access token -> cookie & redirect user to dashboard
      Cookies.set("token", result.access_token);
      this.props.history.push("/dashboard");
    }
    if (result.status == 401) {
      //if error -> set state error
      this.setState({ ...this.state, error: "Invalid username and password." });
    }
  };

  render() {
    const isActive = true;
    return (
      <div
        style={{
          border: "1px solid #c7c7c7",
          width: "500px",
          padding: "20px",
          margin: "auto",
        }}
      >
        <p>{this.state.error}</p>
        <form onSubmit={this.handleLogin}>
          <input
            type="text"
            name="email"
            placeholder="Enter your email address"
            className="input-style"
            onChange={this.handleEmail}
            value={this.state.email}
          />
          <input
            type="text"
            name="password"
            placeholder="Enter your password"
            className="input-style"
            onChange={this.handlePassword}
            value={this.state.password}
          />

          <button style={{ backgroundColor: "tomato" }} id="login-button">
            Login
          </button>

          {/* <button className="login-button">Forgot password</button> */}
        </form>
      </div>
    );
  }
}

export default Login;
