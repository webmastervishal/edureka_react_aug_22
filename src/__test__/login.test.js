import { shallow, mount } from "enzyme";
import Login from "./../pages/Login";
import Dashboard from "./../pages/Dashboard";
import renderer from "react-test-renderer";
import Container from "../components/CurrencyConversion/Container";
import Counter from "../pages/Counter";

describe("Login component", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<Login />);
  });
  test("Number of input boxes to be 2", () => {
    expect(wrapper.find("input").length).toBe(2);
  });

  test("Input should be empty", () => {
    expect(wrapper.find("input[name='email']").props().value).toBe("");
  });

  test("Login button", () => {
    expect(wrapper.find("button").text()).toBe("Login");
  });

  test("snapshot matching", () => {
    const tree = renderer.create(<Login />);
    expect(tree).toMatchSnapshot();
  });

  // test("Dashboard", () => {
  //   const dashboardWrapper = mount(<Counter />);
  //   console.log(dashboardWrapper.debug());
  // });
});
