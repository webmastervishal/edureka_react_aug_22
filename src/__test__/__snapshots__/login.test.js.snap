// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component snapshot matching 1`] = `
<div
  style={
    Object {
      "border": "1px solid #c7c7c7",
      "margin": "auto",
      "padding": "20px",
      "width": "500px",
    }
  }
>
  <p>
    
  </p>
  <form
    onSubmit={[Function]}
  >
    <input
      className="input-style"
      name="email"
      onChange={[Function]}
      placeholder="Enter your email address"
      type="text"
      value=""
    />
    <input
      className="input-style"
      name="password"
      onChange={[Function]}
      placeholder="Enter your password"
      type="text"
      value=""
    />
    <button
      id="login-button"
      style={
        Object {
          "backgroundColor": "tomato",
        }
      }
    >
      Login
    </button>
  </form>
</div>
`;
